sharding/nodesCoordinator/indexHashedNodesCoordinator.go

```diff
@@ -149,10 +145,14 @@ func NewIndexHashedNodesCoordinator(arguments ArgNodesCoordinator) (*indexHashed
 		isFullArchive:                   arguments.IsFullArchive,
 		enableEpochsHandler:             arguments.EnableEpochsHandler,
 		validatorInfoCacher:             arguments.ValidatorInfoCacher,
-		stakingV4Step2EnableEpoch:       arguments.StakingV4Step2EnableEpoch,
 		nodesCoordinatorRegistryFactory: arguments.NodesCoordinatorRegistryFactory,
 	}
 
+	err = ihnc.updateEnableEpochsHandler(ihnc.currentEpoch)
```

Comment:
I'm honestly not sure if this call in constructor is ok. Should ask @AdoAdoAdo's opinion as well

Reply:
the reason behind this was that considering we reach this point with a fresh instance of `enableEpochsHandler`, that never got `EpochConfirmed` called, will have all flags set to false, even though they have epoch 0 for activation. With this call we make sure that this component starts with the flags preset for current epoch